### ISSUE TITLE
Pair-to-Mac : Building iOS app fails on VS2022_ARM to M2 Mac

### ISSUE THREAD

**Issue by baskren** [unoplatform/uno#14466]
### Current behavior

When building a bare bones Uno Project on VS2022 ARM paired to a M2 Mac, building iOS application fails with the following message in the build logs:
```text
2>Tool xcrun execution finished (exit code = 1).
2>ld: warning: ignoring duplicate libraries: '-lSystem.IO.Compression.Native', '-lSystem.Native', '-lSystem.Net.Security.Native', '-lSystem.Security.Cryptography.Native.Apple', '-lcompression', '-lmono-component-debugger', '-lmono-component-diagnostics_tracing', '-lmono-component-hot_reload', '-lmonosgen-2.0', '-lxamarin-dotnet-debug'
2>ld: building for 'iOS-simulator', but linking in dylib (/Users/ben/Library/Caches/Xamarin/mtbs/builds/ProgressRingDemo.Mobile/485f98375c6e0dafb23117e02556c5abca8560a1ed4aaa240781d3a0cb44c32e/C:/Users/ben/.nuget/packages/skiasharp.nativeassets.ios/2.88.1-preview.79/runtimes/ios/native/libSkiaSharp.framework/libSkiaSharp) built for 'iOS'
2>clang: error: linker command failed with exit code 1 (use -v to see invocation)
2>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\16.4.7125\targets\Xamarin.Shared.Sdk.targets(1276,3): error : clang++ exited with code 1:
2>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\16.4.7125\targets\Xamarin.Shared.Sdk.targets(1276,3): error : ld: warning: ignoring duplicate libraries: '-lSystem.IO.Compression.Native', '-lSystem.Native', '-lSystem.Net.Security.Native', '-lSystem.Security.Cryptography.Native.Apple', '-lcompression', '-lmono-component-debugger', '-lmono-component-diagnostics_tracing', '-lmono-component-hot_reload', '-lmonosgen-2.0', '-lxamarin-dotnet-debug'
2>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\16.4.7125\targets\Xamarin.Shared.Sdk.targets(1276,3): error : ld: building for 'iOS-simulator', but linking in dylib (/Users/ben/Library/Caches/Xamarin/mtbs/builds/ProgressRingDemo.Mobile/485f98375c6e0dafb23117e02556c5abca8560a1ed4aaa240781d3a0cb44c32e/C:/Users/ben/.nuget/packages/skiasharp.nativeassets.ios/2.88.1-preview.79/runtimes/ios/native/libSkiaSharp.framework/libSkiaSharp) built for 'iOS'
2>C:\Program Files\dotnet\packs\Microsoft.iOS.Sdk\16.4.7125\targets\Xamarin.Shared.Sdk.targets(1276,3): error : clang: error: linker command failed with exit code 1 (use -v to see invocation)
2>Done building project "ProgressRingDemo.Mobile.csproj" -- FAILED.
```

**NOTE**

- As a POC, MAUI does not have this issue

### Expected behavior

Apps targeting a Pair-to-Mac iOS simulator will successfully build and run, no matter if the VisualStudio 2022 machine is x64 or Arm64, nor if the Mac is an Intel Mac or an Apple silicon Mac.

### How to reproduce it (as minimally and precisely as possible)

- update VisualStudio 2022 ARM to version 17.8.0
- update uno-check to version 1.17.0
- run uno-check
- install Uno Platform VisualStudio Extension, version 5.0.11.16
- In VisualStudio 2022, `File >  New > Project`
- Select `Uno Platform App` in **Create a new project` dialog
- Select `Blank` startup type in the **Uno Platform Wizard** and tap the [Create] button
- Start VisualStudio's **Pair to Mac** and wait for connection to complete
- Switch the **Startup project** to _your_apps_name_.Mobile
- Set the targeted framework to `net7.0-ios`
- Set the targeted device to an iOS simulator
- build and run the application

**Compressed Simple Demo Solution**
[UnoApp1.zip](https://github.com/unoplatform/uno/files/13404276/UnoApp1.zip)

### Workaround

n/a

### Works on UWP/WinUI

Yes

### Environment

Uno.WinUI / Uno.WinUI.WebAssembly / Uno.WinUI.Skia

### NuGet package version(s)

```xml
    <PackageVersion Include="Microsoft.Extensions.Logging.Console" Version="7.0.0" />
    <PackageVersion Include="Microsoft.Windows.Compatibility" Version="7.0.5" />
    <PackageVersion Include="Microsoft.Windows.SDK.BuildTools" Version="10.0.22621.756" />
    <PackageVersion Include="Microsoft.WindowsAppSDK" Version="1.4.231008000" />
    <PackageVersion Include="SkiaSharp.Skottie" Version="2.88.6" />
    <PackageVersion Include="SkiaSharp.Views.Uno.WinUI" Version="2.88.6" />
    <PackageVersion Include="Uno.Core.Extensions.Logging.Singleton" Version="4.0.1" />
    <PackageVersion Include="Uno.Extensions.Logging.OSLog" Version="1.7.0" />
    <PackageVersion Include="Uno.Extensions.Logging.WebAssembly.Console" Version="1.7.0" />
    <PackageVersion Include="Uno.Resizetizer" Version="1.2.1" />
    <PackageVersion Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="5.0.48" />
    <PackageVersion Include="Uno.UniversalImageLoader" Version="1.9.36" />
    <PackageVersion Include="Uno.Wasm.Bootstrap" Version="8.0.4" />
    <PackageVersion Include="Uno.Wasm.Bootstrap.DevServer" Version="8.0.4" />
    <PackageVersion Include="Uno.Wasm.Bootstrap.Server" Version="8.0.4" />
    <PackageVersion Include="Uno.WinUI" Version="5.0.48" />
    <PackageVersion Include="Uno.WinUI.Lottie" Version="5.0.48" />
    <PackageVersion Include="Uno.WinUI.DevServer" Version="5.0.48" />
    <PackageVersion Include="Uno.WinUI.Skia.Gtk" Version="5.0.48" />
    <PackageVersion Include="Uno.WinUI.WebAssembly" Version="5.0.48" />
    <PackageVersion Include="Xamarin.Google.Android.Material" Version="1.10.0.1" />
```

### Affected platforms

iOS

### IDE

Visual Studio 2022

### IDE version

VisualStudio 2022 ARM64 version 17.8.0

**Comment by jeromelaban** [comment]
Thanks for the report. 

You'll need to upgrade your dependencies for SkiaSharp to 2.88.6 (your solution is currently using `2.88.1-preview.79`). Is this happening with a default template coming from Uno 5.0, or from an existing app?

**Comment by baskren** [comment]
It is coming from the most recent VisualStudio generated template  (see below).

<img width="949" alt="image" src="https://github.com/unoplatform/uno/assets/2528888/928b0eb1-26b6-49af-90f2-5f578ea8c37b">

Package exploring doesn't work in VS2022 ARM, so I opened the solution up in VSMacArm and found the following:

<img width="473" alt="image" src="https://github.com/unoplatform/uno/assets/2528888/81e184af-c66b-4e86-8186-a41c3e9f0658">

So it appears that Uno.Winui.Lottie v 5.0.48 is pointing to the wrong versions of SkiaSharp?

**Comment by jeromelaban** [comment]
Uno.WinUI.Lottie v5.0.48 is pointing to 2.88.3, but the uno template is created with 2.88.6.

When creating a template with the blank template (or the recommended), SkiaSharp.Skottie is set to 2.88.6:

<img width="999" alt="image" src="https://github.com/unoplatform/uno/assets/5839577/8594ace9-daa2-4164-8ec8-30de42782910">

Which means that for some reason, you're using an older template. This would mean that for some reason, the app that was create is incorrect, or that vsmac does not support CPM (which is more likely).

Could you take a look at your `Directory,Packages.props` and see which version of Uno.WinUI.Lottie is specified?

**Comment by baskren** [comment]
... and THANK YOU  for the work around!

**Comment by jeromelaban** [comment]
Thanks for the update! 

Indeed, net7.0 is not using the right version. Definitely work a fix! And indeed, if there's no package reference, the CPM file will not be used. We can definitely make a fix for that as well.

**Comment by baskren** [comment]
Thanks for the super responsiveness!

**Comment by jeromelaban** [comment]
This was addressed in 5.2 with SkiaSharp updates.
